2.train-VAE/scripts/nbconverted/beta_vae_training.py

```diff
@@ -8,77 +8,259 @@
+# load the data
+data_directory = pathlib.Path("../0.data-download/data")
+dfs = load_train_test_data(data_directory, train_or_test = "all")
+train_init = dfs[0]
+test_init = dfs[1]
+gene_stats = dfs[2]
 
 
 # In[5]:
 
 
-encoder_architecture = [250]
-decoder_architecture = [250]
+# drop the string values
+train_df = train_init.drop(columns= ["DepMap_ID", "age_and_sex"])
+test_df = test_init.drop(columns= ["DepMap_ID", "age_and_sex"])
 
 
 # In[6]:
 
 
+# subsetting the genes 
+# create dataframe containing the 1000 genes with the largest variances and their corresponding gene label and extract the gene labels
+largest_var_df = gene_stats.nlargest(1000, "variance")
+gene_list = largest_var_df["gene_ID"].tolist()
+gene_list
+
+# create new training and testing dataframes that contain only the corresponding genes
+subset_train_df = train_df.filter(gene_list, axis = 1)
+subset_test_df = test_df.filter(gene_list, axis = 1)
+
+
+# In[7]:
+
+
+print(subset_train_df.shape)
+subset_train_df.head(3)
+
+
+# In[8]:
+
+
+print(subset_test_df.shape)
+subset_test_df.head(3)
+
+
+# In[9]:
+
+
+encoder_architecture = []
+decoder_architecture = []
+
+
+# In[10]:
+
+
+# These optimal parameter values were fetched by running "optimize_hyperparameters.py" and then running "fetch_hyper_params.ipynb"
 cp_vae = VAE(
-    input_dim= dependency_df1.shape[1],
-    latent_dim=90,
-    batch_size=32,
-    encoder_batch_norm=True,
-    epochs=58,
-    learning_rate=0.0001,
+    input_dim=subset_train_df.shape[1],
+    latent_dim=10,
+    batch_size=16,
+    encoder_batch_norm=False,
+    epochs=40,
+    learning_rate=0.005,
     encoder_architecture=encoder_architecture,
     decoder_architecture=decoder_architecture,
-    beta=0.06,
+    beta=1.6,
+    lam=0,
     verbose=True,
 )
 
 cp_vae.compile_vae()
 
 
-# In[ ]:
+# In[11]:
+
+
+cp_vae.train(x_train = subset_train_df, x_test = subset_test_df)
+
+
+# In[12]:
+
+
+# display training history
+history_df = pd.DataFrame(cp_vae.vae.history.history)
+
+# save the training history as a .csv
+hist_dir = pathlib.Path("./results/beta_vae_training_history.csv")
+history_df.to_csv(hist_dir, index = False)
+
+
+# In[13]:
+
+
+# plot and save the figure
+save_path = pathlib.Path('../1.data-exploration/figures/training_curve.png')
+
+plt.figure(figsize=(6, 5), dpi = 500)
+plt.plot(history_df["loss"], label="Training data")
+plt.plot(history_df["val_loss"], label="Validation data")
+plt.ylabel("MSE + KL Divergence")
+plt.xlabel("No. Epoch")
+plt.legend()
+plt.savefig(save_path)
+plt.show()
+
+
+# In[14]:
+
+
+cp_vae.vae
+cp_vae.vae.evaluate(subset_test_df)
+
+
+# In[15]:
+
+
+encoder = cp_vae.encoder_block["encoder"]
+decoder = cp_vae.decoder_block["decoder"]
+
+
+# In[16]:
+
+
+data_dir = "../0.data-download/data/"
+datafs = load_data(data_dir, adult_or_pediatric = "all")
+dependency_df = datafs[1]
+sample_df = datafs[0]
+
+
+# In[17]:
+
+
+train_init['train_or_test'] = train_init.apply(lambda _: 'train', axis=1)
```

Comment:
what is this doing? I cannot tell

Reply:
These functions add a new column to a data frame and then add the designated string to every cell in the newly created column. The intent of this new column is to add labels to ID which were training and testing samples so they could be labeled in the heat map.